packages/core/index.d.ts

```diff
@@ -104,8 +104,8 @@ interface BaseHttpResponse {
 
 export interface HttpResponse extends BaseHttpResponse {
   content: string;
-  data?: object;
-  json?: object;
+  data?: any;
+  json?: any;
```

Comment:
No idea what data is - the same as json?
I think json needs to be any, as object doesn't let you access the json properties in the response

Reply:
Data is the parsed response (but plays nicely with multiple data types). See here: 

https://github.com/zapier/zapier-platform/blob/56f1e53629a99297e6eeea9c3b06a1dc47cbc5c6/packages/core/src/http-middlewares/after/prepare-response.js#L44-L48

as for `any` vs `object`, we know that `data` will be some sort of object type and we want to classify it as such. with `any`, we lose all protection. 

A quick example: [Typescript Playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAZgVzMGBeGAKEAuGIBGAVgKbBQA0MAhrlWAJ4CUaAfDAN4BQMPMA8kVJQAdAGti9CFmYB6GTAgALEAgA2AExgB3EACdR3XgJJkxEqVVnzQazTv2UwIGAEt1xKgEJOhniGEAtlQADhhWMMB0AOSwQcF4AG7EuniCZBC+1IEhYTByME4wusTQui5kLuAZmc4APDAAzHnySioaMMm6eplUMPVN+UH0+MSeMAD1ADoA+hiAIwyM0zLjPrwRVbDqVFC96M5UEBwQCMDAJRC4+CAgqh5gjlQBxLilLmAA5gC+maCQN8TCVQgd4YLY7YTHU7nRg-Kr-QHA0HbKjCAAe4Vatg6ui6ulhf1uCJBKPRzWoqlUMBGUEOVGKeDgcBg2AwnE+QA)

Reply:
https://www.bennadel.com/blog/3126-object-access-bracket-notation-vs-dot-notation-with-typescript-in-angular-2-rc-4.htm

Declaring these types as `object` means the dot notation that is currently used heavily breaks.  I wonder if it would make sense to provide a generic for the expected response type?

Reply:
> Declaring these types as object means the dot notation that is currently used heavily breaks

I don't think that's true, seems fine [in this playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgJIBE5jsg3gKGSOQG0BrALmQGcwpQBzAXStvpAfwF998AbCGGQATLHCoYxyALx5CxGAHtFVAEQAjOFFUAaeUQAWEPnxXJVAd0VQ+w1dwDcvBIpDVFAgHSmGAClHYnkqKAJT4Lm4eEN6KfgFwnkYmoc6u7l4+-mIkAOTBOUxhEenRmfG5SaYFYUA).

I think it's important to communicate that `response.data` definitely isn't a `number` or a `string`. It's _something_ wrapped in curly braces, but before you can safely use it, you have to narrow it with the shape of your data. This is all intended behavior.

Reply:
Try with type `object` though - https://www.typescriptlang.org/play?#code/DYUwLgBAJghmMC4IHsBGArEBjSBeCA3gFASkQBmyySARKjAE40A0JZAFiMMNRDQO7IGwKDSIBfANxEiWZADsAzslAA6HgHMAFLHirKyAJSyFytZp1wYqztyMy5SlSHXJtumAG0A5Ae8BdY0czFwsPH1seAOMgA

I think a generic would achieve that same goal of requiring the user to narrow the shape of their data, but it's a little more streamlined than `const data = response.data as ExpectedResponseType`.

Reply:
Sure, but that playground is sort of backwards - `object` is supposed to be a more vague type (and is uncommonly used if you know anything else about the shape of the data already).

I agree that having a generic for the response instead of of having to re-cast the response would be a great improvement!

Reply:
I had the same problem - I am not able to use dot notation to access any elements in the json of a http response if I use the HttpResponse type - where json is type "object". The object type seems really strict, and seems only appropriate for when you know exactly what the object will look like, but a http response json could be anything (pretty much)